server/services/yamlService.js

```diff
@@ -1,15 +1,56 @@
 /* eslint-disable prefer-destructuring */
 const _ = require('lodash');
 const persistentStore = require('../store/persistentStore');
+const generateIdFromVersionedName = require('../utils/operatorUtils').generateIdFromVersionedName;
+
+const createYaml = (packageName, operatorChannel, globalOperator) => {
+  const installYaml = `apiVersion: v1
+  kind: Namespace
+  metadata:
+    name: my-${packageName}
+  ---
+  apiVersion: operators.coreos.com/v1alpha2
+  kind: OperatorGroup
+  metadata:
+    name: operatorgroup
+    namespace: my-${packageName}
+  spec:
+    targetNamespaces:
+    - my-${packageName}
+  ---
+  apiVersion: operators.coreos.com/v1alpha1
+  kind: Subscription
+  metadata:
+    name: my-${packageName}
+    namespace: my-${packageName}
+  spec:
+    channel: ${operatorChannel}
+    name: ${packageName}
+    source: operatorhubio-catalog
+    sourceNamespace: olm`;
+
+  const globalInstallYaml = `apiVersion: operators.coreos.com/v1alpha1
+  kind: Subscription
+  metadata:
+    name: my-${packageName}
+    namespace: operators
+  spec:
+    channel: ${operatorChannel}
+    name: ${packageName}
+    source: operatorhubio-catalog
+    sourceNamespace: olm`;
+
+  return globalOperator ? globalInstallYaml : installYaml;
```

Comment:
No need to create both const then return just one, check first and create/return the appropriate yaml.

Reply:
This is exact same code as before, just extracted to standalone method for code clarity. 
However yes, there could be 2 separate methods

Reply:
understood but looking at it now, seems like putting the conditional before the creation of the constants would be more efficient.

Reply:
sure